sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -559,8 +559,10 @@ impl TraitMap {
                 suffix: map_trait_name.suffix.0.clone(),
                 is_absolute: map_trait_name.is_absolute,
             };
-            if &map_trait_name == trait_name && are_equal_minus_dynamic_types(type_id, *map_type_id)
-            {
+            let are_same_name_traits = (trait_name.prefixes.is_empty()
+                && map_trait_name.suffix == trait_name.suffix)
```

Comment:
It seems that this would allow for false positives between traits with the same suffix and different prefixes. Some local `MyAdd` and a different `my_library::MyAdd`.

Reply:
In the most recent version it no longer compares suffixes instead before calling `get_methods_for_type_and_trait_name` it converts the trait name call path to contain prefixes when it is possible.